massa-factory-worker/src/denunciation_factory.rs

```diff
@@ -0,0 +1,317 @@
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::thread;
+
```

Comment:
interesting question: could we completely remove de denunciation_factory and only have the denunciation pool in which the state for each DenunciationKey is either `Accumulating(DenunciationData)` or `Denounced(Denunciation)`. The pruning is the same, and when the module is asked for denunciations to include in a block, it only filters out the ones that are ready (Denounced)

Reply:
I'm in favor of merging it as it is and remove the DenunciationFactory when we add the Denunciation in Block headers (next PR) so we have the whole picture in mind